test/src/code_field/code_controller_tab_test.dart

```diff
@@ -68,10 +90,50 @@ public class MyClass {
           modifiers: [const TabModifier()],
         );
         expect(
-          _areIdenticalTexts(controller.text, snippetWithTripleSpaces),
+          areIdenticalTexts(controller.text, snippetWithTripleSpaces),
           true,
         );
       },
     );
+
+    test('works with several tab insertion', () {
+      final controller = CodeController(
+        text: snippetWithTabs,
+        language: java,
+        modifiers: [const TabModifier()],
+      );
+      const tabCount = 2;
+      controller.value = TextEditingValue(
+        text: snippetWithTabs + '\t' * tabCount,
+        selection: const TextSelection.collapsed(offset: 0),
+      );
+      expect(
+        areIdenticalTexts(
+          controller.text,
+          snippetWithDoubleSpaces +
+              ' ' * controller.params.tabSpaces * tabCount,
+        ),
+        true,
+      );
+    });
+
+    test('works with insertion code containing named blocks', () {
+      final controller = CodeController(
+        text: snippetWithTabs,
+        language: java,
+        modifiers: [const TabModifier()],
+      );
+      controller.value = const TextEditingValue(
+        text: '$snippetWithDoubleSpaces\n$namedSectionsWithTabs',
+        selection: TextSelection.collapsed(offset: 0),
+      );
+      expect(
```

Comment:
Please also test the selection preservation.
Please also test the case when the selection was after the tab characters.